keyboards/gmmk/pro/ansi/keymaps/mackrusing/keymap.c

```diff
@@ -0,0 +1,84 @@
+        KC_GRV,  KC_1,    KC_2,    KC_3,     KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,          KC_DEL,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,     KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_PGUP,
+        MO(1),   KC_A,    KC_S,    KC_D,     KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,           KC_PGDN,
+        KC_LSFT,          KC_Z,    KC_X,     KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, KC_UP,   KC_END,
+        KC_LCTL, KC_LGUI, KC_LALT,                             KC_SPC,                             KC_RALT, KC_RGUI, MO(2),   KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+    [1] = LAYOUT(
+        _______, KC_MRWD, KC_MPLY, KC_MFFD,  _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_SLEP,          _______,
+        _______, _______, _______, _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______,  _______, _______, _______, _______, _______, _______, _______, _______,          _______,          _______,
+        KC_CAPS,          _______, _______,  _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______,
+        _______, _______, _______,                             _______,                            _______, _______, _______, _______, _______, _______
+    ),
+
+    [2] = LAYOUT(
+        _______, _______, _______, _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        RGB_TOG, RGB_VAD, RGB_VAI, RGB_RMOD, RGB_MOD, _______, _______, _______, _______, _______, _______, _______, _______, RESET,            _______,
+        _______, RGB_HUD, RGB_HUI, RGB_SAD,  RGB_SAI, _______, _______, _______, _______, _______, _______, _______,          _______,          _______,
+        _______,          _______, _______,  _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______,
+        _______, _______, _______,                             _______,                            _______, _______, _______, _______, _______, _______
+    ),
+
+};
+
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (clockwise) {
+        tap_code(KC_VOLU);
+    } else {
+        tap_code(KC_VOLD);
+    }
+    return true;
```

Comment:
```suggestion
    return false;
```
Returning `true` will also run `encoder_update_kb`, it isn't implemented for this keyboard yet but it's possible it may be added in the future resulting in odd encoder behaviour.